### PR TITLE
Parcel fixes

### PR DESCRIPTION
**Pull request by HypnoMama**
- fixes bugs related to coordinates not being re-set 
- clean up of resetting opportunities
- map now draws both Polygon and MultiPolygon 
- parcel geometry field is now a general Geometry field and will hold either Polygon or MultiPolygon or other potential geometries
- blocking continue/has approval flow fixed for multiple address/parcel/coord changes

### CODE REVIEW

File: cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js [hunk 1]
```diff
@@ -145,24 +148,17 @@ export default function AddOpportunity() {
...
-    dispatch(setGeometry(null));
-    dispatch(setParcelSize(null));
-    dispatch(setPID(null));
-    dispatch(setAddress(null));
+    dispatch(resetOpportunity());
     setError("");
     try {
       const data = await getAddressData(addy);
+      setNoAddressFlag(false); // does this cause an issue?  Shouldn't as SITEID is null at this point right?
       dispatch(setAddress(data.data.features[0].properties.fullAddress));
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/components/Page/AddOpportunity/AddOpportunity.js:162]:
```suggestion
      const data = await getAddressData(addy);
      setNoAddressFlag(false);
      dispatch(setAddress(data.data.features[0].properties.fullAddress));
```

File: cit3.0-web/src/store/reducers/opportunity.js [hunk 2]
```diff
@@ -87,14 +87,18 @@ export default function opportunity(
       state.siteInfo.PID.value = action.payload;
       break;
     case ADD_GEOMETRY:
+      console.log("IN GEOM - payload", action.payload);
+      console.log("current geom: ", state.siteInfo.geometry.coordinates);
       if (!action.payload) {
+        console.log("in geom: should set to null");
         state.siteInfo.geometry.coordinates = null;
+        console.log(state.siteInfo.geometry.coordinates);
       } else if (!state.siteInfo.geometry.coordinates) {
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/store/reducers/opportunity.js:96]:
```suggestion
      if (!action.payload) {
        state.siteInfo.geometry.coordinates = null;
      } else if (!state.siteInfo.geometry.coordinates) {
```

File: cit3.0-web/src/store/reducers/opportunity.js [hunk 3]
```diff
@@ -87,14 +87,18 @@ export default function opportunity(
...
-        state.siteInfo.geometry = {
-          ...state.siteInfo.geometry,
-          ...action.payload,
-        };
+        console.log("in GEOM: else if");
+        state.siteInfo.geometry.coordinates = action.payload.coordinates;
+        state.siteInfo.geometry.type = action.payload.type;
       } else {
+        console.log("in geom ELSE");
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/store/reducers/opportunity.js:101]:
```suggestion
        state.siteInfo.geometry.coordinates = action.payload.coordinates;
        state.siteInfo.geometry.type = action.payload.type;
      } else {
```

File: cit3.0-web/src/store/reducers/opportunity.js [hunk 4]
```diff
@@ -114,9 +118,13 @@ export default function opportunity(
...
       break;
     case ADD_PARCEL_SIZE:
-      if (state.siteInfo.parcelSize.value && action.payload) {
+      console.log("parcelSize: ", action.payload);
+      if (state.siteInfo.parcelSize.value && action.payload !== null) {
+        console.log("add parcel size if");
         state.siteInfo.parcelSize.value += action.payload;
       } else {
+        console.log("add parcel size else");
+
```
Comment (BrendanBeachBC) [inline cit3.0-web/src/store/reducers/opportunity.js:127]:
```suggestion
      if (state.siteInfo.parcelSize.value && action.payload !== null) {
        state.siteInfo.parcelSize.value += action.payload;
      } else {
```

**Review by BrendanBeach — COMMENTED** [review]
Looks mostly good, one question about one of the code comments and a couple of console.logs to remove

**Review by BrendanBeach — APPROVED** [review]
Looks great!

**Review by zacpez — APPROVED** [review]

**Pull request merged**